### ISSUE TITLE
[Announcement] YTM Desktop V2

### ISSUE THREAD

**Issue by Alipoodle** [ytmdesktop/ytmdesktop#1139]
# Hey Everyone 
I'd like to announce some news!

Myself and @NovusTheory are planning on taking over the YTM Desktop project a little more.
and with that we'd like to announce

# YTM Desktop V2.
This is a full re-work of the YTM Desktop Codebase, and providing a fresh integration and workflow for the entire project.
The application is running on a newer version of Electron, running a new Vue Settings and Windows border, written in Typescript, and improving many of the integrations inside of YTM Desktop already has.

**We've currently got the following features:**
- Easy install
- Multiplatform setups
- In-app Navigation [This is for the new Layout]
- Hide to Tray
- Start on boot
- Continue where you left off
- Discord Rich Presence
- Custom Shortcuts for Standard Controls + Volume
- Taskbar Progress + Buttons (Windows only)

**New Features:**
- Always display Volume
- Sleep Timer (this was present in the Beta builds)
- **NEW** Companion Server
  - *This does mean that existing integrations with the Classic YTM Desktop will not function anymore. We considered making it backwards compatible but we'd rather move forward with this release over the previous version due to a number of issues regarding the functionality it provided.*

**However this also means**
That some features have not yet been ported over yet, including:
- Locales
- Mini Player
- Last.FM integration
- Styling YTM With custom CSS
- Audio Output selector
- Audio Patch to set to dB instead of % (Beta Builds)
- Surround Sound / Speaker fill
- URI Protocol
- Dynamic Album art Theming (Beta Builds)
- **AND MANY MORE 😅**

I should note that we do not plan to port over EVERY feature which existed within YTMD currently and these may be due to various reasons.
Obviously if other members are willing to re-introduce these again then we are willing to consider Pull requests for them.

In the mean time we're chugging along in the background preparing the build to a ready state for people to start enjoying the features.
We want to get this out as soon as possible to fix the number of issues which are already in the currnet build like Google Sign in, Missing Navigation, Buttons not working ETC.

**Comment by IndefiniteBen** [comment]
Cool! Excited to see this further improved. 

Some questions: 
1. Are the features listed that haven't been ported yet, what needs to be done before release, or the features that will be missing from v2.0?
2. Am I correct in assuming this means no more releases of V1?
3. When is last.fm support coming? :P I won't install V2 for anything other than some testing until that has been ported (though maybe that needs a full rewrite considering past issues?)

**Comment by Alipoodle** [comment]
1) We've not set a Time or a Minimuim requirement for Features before releasing the first version of v2 yet.
  We're looking to just sort out as much as we can for the time being, prep other things like Auto-updates for existing YTM Desktop will work, Website updates ETC can roll out generally quickly together.
  
2) At **most** there will be 1 more version of v1.14 which will be a "repairs patch" for the Navigation and any other minor features which has been added since v.14.2's release, however given that Development for it is a rollercoast in having everything setup, I suspect that this isn't priorty for myself or Novus at the moment.

3) Last.fm support is one of the more core features on my radar,
  but as all developers giving ETAs:
![SoonTM](https://media.tenor.com/a7aeZxrntk8AAAAC/intensifies-sooning.gif)


We want to begin the project successfully with the correct "new" features and upgrades over the original version, and some of the core features which are quoted and used.

**Comment by realSnosh** [comment]
Can you maybe add a function to add ~ad blockers~ plugins?

Or any interface to make custom addons/plugins?

**Comment by NovusTheory** [comment]
> Or any interface to make custom addons/plugins?

@realSnosh This is something that's been asked a lot for V1 and all I can say is we are talking about it. This will likely be further out than the initial release of V2 (2.0.0) so I would not expect it at that time.

**Comment by kanjurer** [comment]
Will the download functionality be available?

**Comment by Alipoodle** [comment]
> Will the download functionality be available?

Downloading **internally** to YouTube Music is a feature which YouTube will have to do themselves.
Currently they have implemented something on YouTube so they have the knowledge on how to do this safely.

Downloading **externally** to YouTube Music via something like yt-dl is a feature we will not natively include and maybe possible via a plugin although as mentioned not a feature we will have ready for a first release.

**Comment by MrKumaPants** [comment]
Would be willing to help out if y'all need help. I'm a senior dev and have been using React/Typescript for 6+ years.

**Comment by kanjurer** [comment]
I would like to help to if needed :)
I am not very smart though 😂 but would love to contribute ❤️

**Comment by Alipoodle** [comment]
Things are currently a lot changes towards how the internal APIs are working from the Integrations, to getting information from YTM, and more.
For the time being we want to finalised a lot more of the bits before a PR gets made for a lovely feature before it's instantly broken with another change. (This has already happened)
![image](https://github.com/ytmdesktop/ytmdesktop/assets/17199186/d0587938-aa7a-44c0-8bae-6c9178bd3481)

**Comment by marbrink** [comment]
For me my most wanted feature is to be able to sort the songs in my playlists (by artist, title, time it was put in the playlist) ascending and descending.
This seems to me like some really basic functionality that shouldn't be hard to implement from Googles side and should be common sense IMO.

However is there a chance to do this in YTMD?

**Comment by Alipoodle** [comment]
Due to Lazy Loading I suspect that this won't be something we can do properly.
We can only do partical sorting of the rows if we did it, but this kinda isn't a proper solution.

**Comment by kanjurer** [comment]
It it possible to improve YT's shuffle algorithm?

**Comment by IndefiniteBen** [comment]
Another request: better handling of hiDPI screens/switching between them and normal ones. Frequently it gets to a state where the queue is completely gone due to what looks like scaling issues.

**Comment by Alipoodle** [comment]
> It it possible to improve YT's shuffle algorithm?
Not really, again this is a YouTube API which is doing it, and not exactly something which we're able to access all your preferences to make our own Shuffle Algo~
Personally I've not got any issues with it, it does obviously try and priorities adding in 1/2 more songs from the same artist in again.

> Another request: better handling of hiDPI screens/switching between them and normal ones. Frequently it gets to a state where the queue is completely gone due to what looks like scaling issues.
Not quite sure I understand what you mean.

Moving from a 4k -> 1080p or something is a problem at the core of Electron/Chromium and unlikely to be something we can fix.
It does a decent job of trying to reduce the size once it feels like the majorirty is on there.
If you are worthwise talking about scaling inside of YouTube Music, you can make some custom CSS to pick how you want it to work yourself, but obviously or picking to Zoom out of YouTube a bit.

**Comment by IndefiniteBen** [comment]
> Not quite sure I understand what you mean.  

Same. Some combination of different screens with different scaling ratios which are disconnected and reconnected, leads to my problem. If it's still present in v2 I'll try to find the cause and make a bug report 😆 Only the up next/queue panel has the issue.

> Moving from a 4k -> 1080p or something is a problem at the core of Electron/Chromium and unlikely to be something we can fix.

I'm going to assume this is the core of the problem so won't be fixed. That's good to know, thanks!

**Comment by Caboose12000** [comment]
Will V2 work on linux out of the box, or will I need to use wine/proton for a while at first? (can't wait for the new release! 😁)

**Comment by DaviMedrade** [comment]
If V2 works on Linux right now (I haven't tested it yet), does it support MPRIS? If not, is that a planned feature?

**Comment by Alipoodle** [comment]
> Will V2 work on linux out of the box, or will I need to use wine/proton for a while at first? (can't wait for the new release! 😁)

I've made myself a build inside of WSL (Windows Subsystem for Linux) and we have builds for Linux building fine. So I see no issue as to why this wouldn't work on Linux (I was aware that Sandboxing was a problem on the current v1.13/v1.14 builds.)

_I should note that in WSL I ran it without GPU but this is likely a problem with how WSL works rather than Linux having issues_


> If V2 works on Linux right now (I haven't tested it yet), does it support MPRIS? If not, is that a planned feature?

Currently no work has been made for MPRIS although this shouldn't be to difficult of a thing to impliment with the APIs which we already have for YTMD to use in other Integrations we have made (like Discord.js / Last.fm ETC)
No garantee that this will be done for the initial release for V2 however I should feel that this shouldn't be too far off the list for re-adding.

Obviously the majority of OS users (and in turn our downloads for the software) come from Windows, and therefore Universal + Windows specific functions have been generally higher priority for us to get working on.
_and also lacking a Mac to try running MacOS specific features at all_

**Comment by stex** [comment]
> Where is the V2 source? Github?

My guess would be this branch: https://github.com/ytmdesktop/ytmdesktop/tree/v2

I just built it for macOS (arm64) since the 1.x version would not let me log in as Google deemed it an unsafe browser.

Login and playback work like a charm, the settings button opens an empty dialog. Looking forward to the next merges! 🙂

**Comment by TheDavidJohnson** [comment]
Jumping in to say I love what you've done & you're doing, and happy to test anything at any point in time! Using the app on Linux (Ubuntu Desktop) and MacOS (Apple silicon / M1). Not a true dev, but a PM and not afraid to run stuff that breaks if I can help!

**Comment by glebarr** [comment]
how do I remove ytmdesktop?

**Comment by rd-626** [comment]
@glebarr first of all, what method u used to install it?

**Comment by glebarr** [comment]
I installed it thru the terminal, but couldn't uninstall with the same method

**Comment by rd-626** [comment]
@glebarr I can't help you with out the details....... What OS? And which command did you use to install the program?

**Comment by glebarr** [comment]
Linux, don't recall the exact command, but I just copied and pasted the terminal commands as instructed; pretty standard really

**Comment by rd-626** [comment]
Oh nice

**Comment by Alipoodle** [comment]
> > Where is the V2 source? Github?
> 
> My guess would be this branch: https://github.com/ytmdesktop/ytmdesktop/tree/v2
> 
> I just built it for macOS (arm64) since the 1.x version would not let me log in as Google deemed it an unsafe browser.
> 
> Login and playback work like a charm, the settings button opens an empty dialog. Looking forward to the next merges! 🙂

Have you tried the V2 branch yet?
The v1 branch to my knowledge never had M1 support? (At least proper arm support, still worked in Rosetta)
A few MacOS changes went in recently to fix some *stupid* bugs.

**Comment by NovusTheory** [comment]
## 🎉 Exciting News: Release candidate build for YouTube Music Desktop v2!
This is an early access version of YTM Desktop v2 to gather feedback of the changes we have done and public testing before pushing a full release.

We've tried our best to improve the existing features to function fully and introduce a few new features here and there. As mentioned we've not included everything from the original version, and future versions may see them again.

We've done most of our work on Windows, and poked a little bit at MacOS & Linux to make sure that things are working on them and are hoping to release the full version of v2 soon, however we want to make sure that things work and are bug free.

It is safe to assume that all features are now frozen and we are not looking to implement any new major features until another release after v2. If anything is missing that seems pretty important we will evaluate it before releasing v2 otherwise it should be mostly bug fixes and small quality of life changes.

To obtain this release you can go to https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.1

<sub>*Message half written by @Alipoodle*</sub>

**Comment by GallopingCow** [comment]
This makes me so happy. I'll be waiting with baited breath for a mini player.
I do feel a little weary about entering my Google Password into "random" apps though, since everywhere else I can sign in with a phone prompt. Not being able to verify a legit browser sign-in page feels bad. :(

**Comment by Caboose12000** [comment]
I just installed v2 and I love the new UI!  Will the linux version be available as a flatpak?  =)

I run Kubuntu 23.04 and I noticed a couple things that didn't work right away for me, not a big deal or anything i just figured I'd mention it here since you guys asked for feedback:

- Discord rich presence doesn't seem to work for me, it's like discord isn't detecting ytm at all (I have discord v0.0.28 from flathub)
   - it seems like the settings > integrations > discord rich presence toggle doesn't save/update, it's always off when I leave settings and come back
- I tried to bind F1 and F7 to volume up and volume down and they didn't save / reset just like discord rich presence
- toggling "Show track progress on taskbar" doesn't seem to do anything (although the toggle does remember it's state between settings menu sessions)
- When I click the tray icon for more info it doesn't load album art (I'm not sure if it's supposed to or not, but there seems to be a spot to display it)
- I closed the app, then re-opened it, and the play button disappeared or failed to load.  It reappeared when I clicked the space it usually occupies, like it was just invisible at first


That's everything I've noticed so far, hopefully that's helpful to you guys!  I hope it doesn't come off as complaining 😬 I'm absolutely thrilled to have a linux version of v2 right out the gate at all, let me know if there are any logs or other info that'd be helpful to include!

**Comment by e2t** [comment]
Thank you for the new version of the program! 
There is a small problem with hotkeys. I set the hotkey (Ctrl+Alt+P) to play/pause and it didn't work. I removed the hotkey (None), and now the previous combination works. A little weird, but I like it))

**Comment by Alipoodle** [comment]
>I do feel a little weary about entering my Google Password into "random" apps though, since everywhere else I can sign in with a phone prompt. Not being able to verify a legit browser sign-in page feels bad. :(

We sadly cannot use a OAuth2 Sign in for this due to being a Google provided product and not a site. The full source code is avalaible for you to review and check in terms of the security of your account. Build for youself if you feel like we might have bugged the release versions or whatever. Full development guide is in the Read Me (in v2 branch)

With the updated version of Electron you should be able to:
Use PassKey (This prompts ONLY as a Password alternative) and use the full suite of 2FA Methods (besides the Security Key due to missing extension in Electron).

> There is a small problem with hotkeys. I set the hotkey (Ctrl+Alt+P) to play/pause and it didn't work. 
Strange that re-adding it does work. I see we have an issue for it now so will do some investigation soon into what might be going wrong.

**Comment by Alipoodle** [comment]
> Will the linux version be available as a flatpak?

https://www.electronforge.io/config/makers/flatpak 😄 
We haven't set it up just yet but should be a super easy install for getting a `.flatpak` file to export.


> Discord rich presence doesn't seem to work for me

**Discord is very unsupported on Linux with only minor support for keeping basic functionality. 😓** 
[*For Linux: Because of the variations in both software and hardware for computers running Linux, technical support for this operating system is limited.*](https://support.discord.com/hc/en-us/articles/213491697-What-are-the-OS-system-requirements-for-Discord-#docs-internal-guid-09be9a94-7fff-2c2f-1ad4-48886516463f)
I'm not aware of it entirely having Discord rpc broken, It shouldn't disable itself so might have to investigate what's going on with that, but after failing it should show a retry button
![image](https://github.com/ytmdesktop/ytmdesktop/assets/17199186/bbfdab0e-1b92-4d42-a7d2-42bab49b496d)

Can I confirm that you are not having issues with other Discord RPC applications and this is just a YTMD issue?


> Toggle "Show track progress on taskbar" doesn't seem to do anything

Does this work on a restart? If not will have to investigate, this was known working on MacOS and Windows but Linux is a wild space for different enviroments and how Progress might be shown for that.


> When I click the tray icon for more info it doesn't load album art

Can you provide some more info for this, The tray Icon shouldn't display Album Art or anything? 🤔 
The tray icon should:
- Left Click open
- Right click show context menu for
![image](https://github.com/ytmdesktop/ytmdesktop/assets/17199186/550622b2-b8e5-418a-bdc1-6609efb63c20)


> I closed the app, then re-opened it, and the play button disappeared or failed to load. It reappeared when I clicked the space it usually occupies, like it was just invisible at first

This is a bug with how we use the Continue where I left off + Pause on start up.
Last minute we improved how the Pause on startup works by setting it to *Require user input* to play, however occationally this causes YouTube Music to fail to display the play button... 😓 
Previously we *fixed* this by playing and pausing the player quickly, however this tended to cause the music to play for a second and not as ideal.

**Comment by Caboose12000** [comment]
> We haven't set it up just yet but should be a super easy install for getting a .flatpak file to export.  

Super glad to hear it!

> Discord is very unsupported on Linux with only minor support for keeping basic functionality. 😓  

I understand, I signed up to be in a small niche when I started using linux, I honestly just really appreciate having a linux version of YTMD at all 😁

> I'm not aware of it entirely having Discord rpc broken, It shouldn't disable itself so might have to investigate what's going on with that, but after failing it should show a retry button

I never saw a retry button or connection failed message
![DiscordRpcToggleNoRetry](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/2adb558f-61ef-4d88-b35a-cbd8bf19d208)


> Can I confirm that you are not having issues with other Discord RPC applications and this is just a YTMD issue?

The Dawncraft modpack for minecraft is the only other rpc app I have right now, here's a screenshot of it working on my system
![DiscordRpcOnLinux](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/b3e064fc-6557-496a-9839-fb1c806e21b1)


> Does this ["Show track progress on taskbar" toggle] work on a restart? If not will have to investigate, this was known working on MacOS and Windows but Linux is a wild space for different enviroments and how Progress might be shown for that.  

Even after a restart, it still doesn't work for me as far as I can tell.  But honestly between KDE, Gnome, Xfce, etc, I wouldn't blame you guys if you decided to just not support this feature on linux, it sounds like a real headache lol
![NoShowTrackProgress](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/114d0d15-b27d-4306-9d63-3e5e7ebeabdc)

> Can you provide some more info for this, The tray Icon shouldn't display Album Art or anything? 🤔

I'm kind of embarrassed about this one 😅 I accidentally confused KDE's built in media player tray icon for the new YTMD tray icon, the YTMD tray icon works exactly as you described, while the KDE media icon menu looks like this (sorry!):
![KDEMediaTrayMenu](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/e19c2134-9df6-442e-849c-6328a2d27885)


> This is a bug with how we use the Continue where I left off + Pause on start up.
Last minute we improved how the Pause on startup works by setting it to Require user input to play, however occationally this causes YouTube Music to fail to display the play button... 😓
Previously we fixed this by playing and pausing the player quickly, however this tended to cause the music to play for a second and not as ideal.

I definitely prefer the solution you guys went with then!  


Thanks for taking time to respond to and consider everything from my first message!  Sorry to leave another wall of text for you 😅

**Comment by Alipoodle** [comment]
> I'm kind of embarrassed about this one 😅 I accidentally confused KDE's built in media player tray icon for the new YTMD tray icon, the YTMD tray icon works exactly as you described, while the KDE media icon menu looks like this (sorry!):

Now this is up to the Linux System, but this might be something which we can ADD to (depends on how it functions).
[MPRIS](https://wiki.archlinux.org/title/MPRIS) is one of the things we are looking to support later down the line to give all the data.
For MacOS + Windows it appears to send the album art to their respected media systems.
![image](https://github.com/ytmdesktop/ytmdesktop/assets/17199186/bf058f22-f165-4af8-83e9-cae40ab60df1)


> Discord Rich Presences + Progress in Taskbar
Obviously things are still in Early access, so please don't lose hope in these not being implimented, these might be something we've messed up with ourselves for how these can communicate with the OS.

**Comment by pixsrv** [comment]
hi there
i've checked your setup.exe file with google virus total and it says this
2 security vendors and no sandboxes flagged this file as malicious
have a look here
https://www.virustotal.com/gui/file/a5c19da10b4108dc8dc46f316bcee4af4d44dd1f265d7110ae992f907d19c9a8

**Comment by Alipoodle** [comment]
First of all, thank you for doing a fair audit of the setups to make sure it's safe.
However I wish for you to believe that we are not doing anything malious here.

A lot of applications can easily get false positives, AVs are designed to be Shoot first, ask questions later as this is a better approach to it all to keep users safe before things go too wrong.

We don't hide our code, it's fully visible at: https://github.com/ytmdesktop/ytmdesktop/tree/v2
We didn't make the build, GitHub actions did, and the full action can be found here: https://github.com/ytmdesktop/ytmdesktop/actions/runs/6030540849
This means we couldn't have injected in any code for that specific build, you can look through the entire source as of that commit to check what we do.
![image](https://github.com/ytmdesktop/ytmdesktop/assets/17199186/fa2e1c19-dd3d-413e-863e-006f211f4fe0)

Our code is not signed due to costing a decent sum of money to get a certifiate for it. This is probably one of the main reasons we are flagged as suspious in most applications.

**Comment by pixsrv** [comment]
when YTMD doesn't respect neither where it was docked (to which screen edge) nor if it was docked at all

the most left edge (of the most left monitor) causes that YTMD escapes from the view port

**Comment by pixsrv** [comment]
shortcuts are not working at all

**Comment by Alipoodle** [comment]
> shortcuts are not working at all

#1150 

> the most left edge (of the most left monitor) causes that YTMD escapes from the view port
Could you share a picture?
This sounds like a VERY old Electron bug which has been Well fixed by now.

**Comment by minipasila** [comment]
> * Taskbar Progress + Buttons (Windows only)

Not sure if this feature is supposed to be fully added yet but for me it only shows the progress bar, but it doesn't show those buttons as shown here:
![image](https://github.com/ytmdesktop/ytmdesktop/assets/4912958/df5f08a2-be9d-4e8e-a220-7c49d7debe8a)
I tried restarting the app and it still does that. I am playing music during that screenshot.

**Comment by gabuch2** [comment]
So far I haven't got any problem aside from minor visual bugs. (Dock icon is black, making it hard to see in the Ubuntu top bar, for example). But what I'm really missing is the select output device option. 🙏 Hoping it will be added soon.

**Comment by NovusTheory** [comment]
> Not sure if this feature is supposed to be fully added yet but for me it only shows the progress bar, but it doesn't show those buttons as shown here

Not sure what might be going on here. Log files and other debug information were added into development and we're working to add debug logging to everything so we can ask for it. Currently nothing like this exists in rc.1.

> (Dock icon is black, making it hard to see in the Ubuntu top bar

Noted, will see what can be done to help alleviate this because everyone has different theme colors etc

> But what I'm really missing is the select output device option

This was something that didn't quite make the cut as the method that is used to do this is very scuffed. While it did work the experience of how it worked was very poor. Unfortunately not every OS has a native audio output selector for apps, such as with Windows so this will be closely monitored as I'd like to have this possible without the scuff.

> Things i wish were available but I seen in a not-done/to-do list:
    Other than opening Chromium's Developer Tools to make use of custom CSS

We'll look into adding the ability to open YTMs devtools. Currently the setup has the main window overriding the keybind for opening devtools.

> Going full-screen (F11) messes with the windows title bar a little bit

Noted, fullscreen likely isn't handled correctly for this case as it is handled correctly if YTM were to fullscreen

**Comment by yochannah** [comment]
Is this the right place to share a bug I'm getting on launch of the x64 mac executable? 

``` Uncaught Exception:
Error: Could not get code signature for running application
at 54575 (/Applications/YouTube Music Desktop App.app/Contents/Resources/app/.webpack/main/index.js:8:640600)
at i (/Applications/YouTube Music Desktop App.app/Contents/Resources/app/.webpack/main/index.js:8:1376119)
at /Applications/YouTube Music Desktop App.app/Contents/Resources/app/.webpack/main/index.js:8:1376891
at Object.<anonymous> (/Applications/YouTube Music Desktop App.app/Contents/Resources/app/.webpack/main/index.js:8:1376918)
at Module._compile (node:internal/modules/cjs/loader:1271:14)
at Object..js (node:internal/modules/cjs/loader:1326:10)
at Module.load (node:internal/modules/cjs/loader:1126:32)
at node:internal/modules/cjs/loader:967:12
at Function._load (node:electron/js2c/asar_bundle:2:13330)
at node:electron/js2c/browser_init:2:122264
```

<img width="514" alt="Screenshot of the above error message, as shown on a mac OSX UI" src="https://github.com/ytmdesktop/ytmdesktop/assets/9271438/4b9068cf-779f-4d24-bb92-08d80828c9f8">

--

This is on an iMac running OSX 13.5.1, can provide more specs if useful! :) 

I'll try compiling locally and see if it's the same.

**Comment by senolfndk** [comment]
add ad blocking option sponsorblock https://sponsor.ajay.app/

**Comment by Alipoodle** [comment]
> add ad blocking option sponsorblock https://sponsor.ajay.app/

We've mentioned this heavily that we have no intention of adding this natively into the application. 
If it's possible via a plugin or something when we make it, we will allow it but likely won't promote the usage it.

**Comment by senolfndk** [comment]
> > reklam engelleme seçeneğini ekleyin sponsorblock https://sponsor.ajay.app/
> 
> Bunu native olarak uygulamamızın eklemediğini çokça belirtmiştik. Bunu yaptığımızda bir eklenti veya başka bir şey mümkünse, buna izin verilmesine rağmen muhtemelen hatırlatma teşviki durdurulmaz. #114

😞

**Comment by BrunoGGM** [comment]
@Alipoodle How can I contribute to the project? I use ytmdesktop constantly and I would like to contribute ♥️

**Comment by Alipoodle** [comment]
> @.Alipoodle How can I contribute to the project? I use ytmdesktop constantly and I would like to contribute ♥️

The current source for v2 is in the v2 branch. We likely won't be adding new features into the initial release of v2 as we've got it to a satisfied place and patched _most_ of the bugs.,
At that stage we will likely be pushing the code into the main branch + development and continuing development from there.

We're entirely open for new features to be added into soruce code (there's lots of things I want to work on to get it closer to the v1 specs)

**Comment by pbldk** [comment]
I get this odd error. Its a great app by the way, works great.
1. the app has run for some time.
2. I close the app while a song is playing, pressing the top right close button
3. The error comes, and I cant close the error window.
4. The song finishes, even though the app has closed.
5. Two youtube apps are running in the tast manager, and an "error", have to close them manually.



![odd_error](https://github.com/ytmdesktop/ytmdesktop/assets/32420497/21d82785-942b-4b22-b372-9f55d7de8f74)

**Comment by NovusTheory** [comment]
## 🎉 v2.0.0-rc.2 release
We have gone over all the feedback that's been provided and fixed a lot of the glaring issues that has been reported and improved some quality of life in the application. For full details and downloads for the v2.0.0-rc.2 go here https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.2

**Comment by IGS-GIT** [comment]
> It it possible to improve YT's shuffle algorithm?

we as a community cannot adjust or rewrite Google's proprietary APIs. That source isn't publicly available, so the only people able to make the shuffle algorithm changes is Team YouTube.

**Comment by pbldk** [comment]
Could it preload the album art and playlist items or could that be an option? It's stuttering a bit when I scroll on a large playlist.
Maybe there could be an option to set a size on the HD for preloaded items. I think iTunes does this.. not sure.

Also, what an amazing work you have done. The app is just 10/10.

**Comment by Alipoodle** [comment]
> Could it preload the album art and playlist items or could that be an option? It's stuttering a bit when I scroll on a large playlist. Maybe there could be an option to set a size on the HD for preloaded items. I think iTunes does this.. not sure.
> 
> Also, what an amazing work you have done. The app is just 10/10.

Considering your playlist can be updated from your phone etc, this isn't exactly an option as what we cache can be wrong 
Best which can be done is changing the caching for the album arts images to store for longer on your system but this feels overkill 😅.

I've heard so many complaints about the poor UX on playlist and how their scrolling works, but they will have to implement something themselves to add something like Skeleton based scrolling.

**Comment by stevenmccormack** [comment]
First of all, thanks for the great work!
Please find below my feedback using rc2 on Debian :
- Please bring back the configurable show/hide window hotkey
- The app shows as Chromium in PulseAudio. As a lot of other apps do as well (PWA for teams, if you use a chromium based browser) you often loose the sound device configuration. When I had a teams meeting and then start the YTM app my headset will play instead of my speakers, when it had been used for the meeting. Maybe it would be possible to save a preferred audio device in YTM.

**Comment by Alipoodle** [comment]
> Show/Hide Key

This is something possible, but might not make it to the release of v2.

> Select Audio device & Chromium in PulseAudio

Selecting Audio device is on our list, but the functionality of it worked before, just didn't work very well (song change generally reset the device back to default)
https://github.com/brave/brave-core/commit/a1c59967b0a8ca65705b9085716b97754ac5f3bd - Checking for similar issues for this, Brave appear to have to edit the Chormium Source to fix their issue... given that we have a pre-compiled one from Electron, I suspect that this may not be something we can fix ourselves without a custom compiled version of Electron 😅
https://github.com/electron/electron/issues/27581 - Given that this is a FR from Feb 2021 and even notes issues from places like **Discord** and **VS Code,** this is a WIDE spread issue and low on their priority list... 😅

**Comment by gotgenes** [comment]
Can you please check the artifact for `YouTube.Music.Desktop.App-darwin-arm64-2.0.0-rc.2.zip`? I'm getting an error,
>"YouTube Music Desktop App" is damaged and can't be opened.

![Screenshot 2023-09-11 at 10 07 36 AM](https://github.com/ytmdesktop/ytmdesktop/assets/8517/32fc3905-d232-4918-a2b7-60897956f5ac)

I have tried re-downloading the artifact and get the same result.

**Comment by NovusTheory** [comment]
> Can you please check the artifact for YouTube.Music.Desktop.App-darwin-arm64-2.0.0-rc.2.zip? I'm getting an error,

We'll look into what's going wrong with the mac builds and try to correct them by rc.3

**Comment by cgalbreath** [comment]
Not sure if this is known but it might be, cannot sign-in to google in this build on Windows 11

**Comment by Alipoodle** [comment]
> Not sure if this is known but it might be, cannot sign-in to google in this build on Windows 11

Which build are you on? (Should show in setting)
And what message appears?

We've not experienced this ourselves.

**Comment by cgalbreath** [comment]
> > Not sure if this is known but it might be, cannot sign-in to google in this build on Windows 11
> 
> Which build are you on? (Should show in setting) And what message appears?
> 
> We've not experienced this ourselves.

There was no error message when I tried to login it just didn't do anything, I will report back if it does this again. 

I had just obtained the latest build and uninstalled the prior version, to clarify this is my first time using this current v2 build.
I kept trying to turn on the Companion Server which seems to not be functional but I will try again now that it is letting me login.

**Comment by gtimbo** [comment]
> Best which can be done is changing the caching for the album arts images to store for longer on your system but this feels overkill 😅.

Are these cached at all now? I mean, I assume they are, but I tend to find a lot of the time taken to load the (stupidly annoying!) progressive lists is showing album art (i.e. somewhere between 1 and 5 seconds, or longer, between the text showing and the images being displayed, depending on what else is going on). Maybe they are cached and it's just the underlying Google code that's loading things in this way -- prioritising text over images, which admittedly does make sense -- but it very much makes the interface feel laggy.

**Comment by Alipoodle** [comment]
> > > Not sure if this is known but it might be, cannot sign-in to google in this build on Windows 11
> > 
> > 
> > Which build are you on? (Should show in setting) And what message appears?
> > We've not experienced this ourselves.
> 
> There was no error message when I tried to login it just didn't do anything, I will report back if it does this again.
> 
> I had just obtained the latest build and uninstalled the prior version, to clarify this is my first time using this current v2 build. I kept trying to turn on the Companion Server which seems to not be functional but I will try again now that it is letting me login.

One issue which I've been reminded off which might be the cause.
If you are playing music the "Login" button won't work as it will prompt you with a "Are you sure you want to leave?" Which fails to show. Because of this it rejects the request silently.

Try seeing if you pause your music and it fixes it~

**Comment by NovusTheory** [comment]
# v2.0.0-rc.3 release (security update)
Thread watchers please update to https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.3. This includes a critical security patch for CVE-2023-4863. There are other changes but this is a security update first. This was released as a way to at least get a patched build out there.

**Comment by zejjnt** [comment]
last.fm isn't working for me, it just stays at "No" for authentication. I am redirected to last.fm and I have authenticated the app through there.
Am I missing something?

**Comment by Alipoodle** [comment]
> last.fm isn't working for me, it just stays at "No" for authentication. I am redirected to last.fm and I have authenticated the app through there. Am I missing something?

We don't authenticate on app side instantly to know if you clicked yes, so the "No" can display for a song or two after originally clicking the authenticate button. 
If you are constantly being promoted to Authenticate with L.fm then there's a problem with the connection

**Comment by zejjnt** [comment]
> > last.fm isn't working for me, it just stays at "No" for authentication. I am redirected to last.fm and I have authenticated the app through there. Am I missing something?
> 
> We don't authenticate on app side instantly to know if you clicked yes, so the "No" can display for a song or two after originally clicking the authenticate button. If you are constantly being promoted to Authenticate with L.fm then there's a problem with the connection

It fixed itself after I had played a track as you said, thanks :)

**Comment by ross8alex** [comment]
Will there be a new version of the remote control app that gets released? The current version seems to be unavailable to new devices because of the Google Play restriction on apps that haven't been updated to target Android 13.

**Comment by sunggooni** [comment]
I cannot see any content in the ytmdesktop v2 window as follows:
![image](https://github.com/ytmdesktop/ytmdesktop/assets/63690358/fd322fc8-c398-497d-b17a-519f01944ae9)

I was trying to run ytmdesktop v2 [v2.0.0-rc.3] (https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.3) on Windows 10 and Ubuntu 22.04.
This happened in the RC.1 and RC.2.
Do I need to install any libraries to run ytmdesktop v2 on Windows 10 and Ubuntu 22.04?

Now I am using ytmdesktop v1 without any problems.

Thanks.

**Comment by mross** [comment]
I fully understand that this might be an edge request, but!
I am really hoping for zoom levels above 150%. I spend a lot of time at a workbench 15 feet away from a 50-inch display, and even at 150% zoom I cannot quite read the track titles, let alone the smaller text elements like time.

Thanks for everything; really looking forward to v2

**Comment by Alipoodle** [comment]
> I cannot see any content in the ytmdesktop v2 window as follows:
> (Image to Blank YTMD V2)

From the issue which ElectricalBoy has opened we have found that in some regions they don't allow you to access YTM without having a premium account.

Could I confirm your location?
And could you open a Incognito/Private window and open https://music.youtube.com/ and see where it takes you? 😄 

> I am really hoping for zoom levels above 150%. I spend a lot of time at a workbench 15 feet away from a 50-inch display, and even at 150% zoom I cannot quite read the track titles, let alone the smaller text elements like time.

One of the things which I'm looking into is to introducing the Zoom level, as I personally run my YTM at 90% most of the time to give me a bit more of a view of everything.
I think I will likely keep it consistent with chrome of a Minimum of 25% and maximum of 300% zoom [Chrome does go up to 500% but I feel like we might be pushing it]

**Comment by Alipoodle** [comment]
> Will there be a new version of the remote control app that gets released? The current version seems to be unavailable to new devices because of the Google Play restriction on apps that haven't been updated to target Android 13.

For the time being myself and Novus lack skills to work on the Android app.
HOWEVER I think the intension is to make a web based version of the app directly implemented inside of Companion server.
This way it's not Android specific and can be made into a PWA app on an Android, iOS, Linux or Windows device without any apps 😋

There are some features might not work so smoothly but we can cheat some bits via the [`mediaSession`](https://developer.mozilla.org/en-US/docs/Web/API/MediaSession) API in Browsers to show the song data in a notification (not sure about length ETC as we might have to have a empty Audio element to "match" current progress and display this.

**Comment by sunggooni** [comment]
> > I cannot see any content in the ytmdesktop v2 window as follows:
> > (Image to Blank YTMD V2)
> 
> From the issue which ElectricalBoy has opened we have found that in some regions they don't allow you to access YTM without having a premium account.
> 
> Could I confirm your location? And could you open a Incognito/Private window and open https://music.youtube.com/ and see where it takes you? 😄
I am also in Korea. When I try to open https://music.youtube.com/ in a Private window, it takes me to https://www.youtube.com/premium.
Thanks.

**Comment by NovusTheory** [comment]
# 🎉 v2.0.0-rc.4 release

This release has many improvements, quality of life changes, and bug fixes. For full details and downloads for v2.0.0-rc.4 go here [v2.0.0-rc.4 (release)](https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.4).

We feel like v2.0.0 is very close and this release candidate could get upgraded to v2.0.0 if all goes well. We thank everyone for all the feedback and bugs reported to make this application the best it can be.

**Comment by Caboose12000** [comment]
Just got rc4, I love the new loading screen!

I know discord itself isn't even supported very well on linux, but I figured I'd report this since it resulted in a crash for me, just so you guys are tracking it:  
  
I'm on Kubuntu 23.04 and when I opened  YTMD settings, Discord rich presence was toggled "On" by default (probably because i left it on in rc3 lol), and I tried to toggle it "Off" and the app crashed
![ytm_discord_crash](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/a1b83776-079e-44dd-8a6e-5a38e3ef8c93)

Let me know if you need any more details on this, otherwise I'm getting really excited for the v2 release!

**Comment by Alipoodle** [comment]
@Caboose12000 This is to do with a missing key in the config file 
Deleting it and YTMD will recreate it with all the values fixed (understandable reset to default values).

Have you tried each of the Release candidates or only some?
As it might have not added/removed the settings correctly 😅

**Comment by Caboose12000** [comment]
@Alipoodle I have tried each rc right when they came out

deleting the config does seem to have fixed it for me, I'm able to toggle discord rich presence now without any crashes!  (discord itself still doesn't acknowledge it, but afaik that's to be expected in linux for now, right?)

**Comment by SomeoneSK** [comment]
Are you planning to add adblock?

**Comment by Alipoodle** [comment]
> Are you planning to add adblock?

Please see #114 which explains our stance on Adblock.

**Comment by ClarkCrichton** [comment]
hey love the program! commenting to ask if yall can add a pitch shifter? i like to  play guitar along with songs on ytmusic and songs tend to be in a bunch of different tunings and i dont wanna retune for each song lol. also a speed thing like 1.25x speed and stuff like whats on youtube itself would be mega helpful too. sorry if this is asking too much, and thanks again for making this app, its really convenient :)

**Comment by NovusTheory** [comment]
Looks like YouTube Music has internally changed some things. No release is being pushed out quite yet but if the app's loading screen is stuck at "Loaded YouTube Music" then you will need to download [this build](https://github.com/ytmdesktop/ytmdesktop/actions/runs/6488833374) from our CI. Discussion is taking place to review if there is a method to allow updating certain scripts independently if YouTube Music were to push changes that break the app such as this one to make it more seamless to fix and deploy.

**Comment by Vexnass** [comment]
Is the "Enable speaker fill" option working as of right now? You've stated that its not ported yet. Option is there to be implemented in the future or its not working as expected?

**Comment by bmcintyre7** [comment]
Are there swagger docs or the like for the companion server api anywhere?

**Comment by Alipoodle** [comment]
> Are there swagger docs or the like for the companion server api anywhere?

https://github.com/ytmdesktop/ytmdesktop/wiki/v2-%E2%80%90-Companion-Server-API-v1

**Comment by Asbaak** [comment]
Would be nice if you could implement the dynamic lyrics like in the YTM App for iOS/Android or like in the Apple Music application.
Added a screenshot of what I mean for better understanding.

Thanks! :)

![photo_5292277931156033539_y](https://github.com/ytmdesktop/ytmdesktop/assets/106897735/7c499a46-d691-447c-a8be-f213e89dafb8)

**Comment by Alipoodle** [comment]
> Would be nice if you could implement the dynamic lyrics like in the YTM App for iOS/Android or like in the Apple Music application. Added a screenshot of what I mean for better understanding.

Real Time / Synced Lyrics is a already requested feature in #809.
Considering we'd need to get a source of data which provides us with timestamps for synced lyrics

**Comment by NovusTheory** [comment]
# 🎉 v2.0.0-rc.5 release

This release fixes the internal changes from YouTube Music. For full details and downloads for v2.0.0-rc.4 go here [v2.0.0-rc.5 (release)](https://github.com/ytmdesktop/ytmdesktop/releases/tag/v2.0.0-rc.5).

**Comment by MartinCura** [comment]
v2 feels so close 🥳 great work you're doing here!!

do you have a timeframe for it? otherwise: i wanted to try out the rc5 but it's not up on AUR yet ([still has rc4](https://aur.archlinux.org/packages/ytmdesktop-git))

side question: did you manage to find a way to survive further internal changes that YouTube Music may do as they did after rc4? Thanks !

**Comment by NovusTheory** [comment]
> do you have a timeframe for it?

We're working on things we want to do before v2 fully releases and waiting on some upstream stuff then we'll probably ship it out. This app is a hobby project so we can't guarantee timeframes.

> side question: did you manage to find a way to survive further internal changes that YouTube Music may do as they did after rc4? 

I did some research and programming to experiment with providing certain scripts dynamically to help prevent this problem but ultimately we're deciding right now to keep that on the back burner and if YouTube Music pushes a breaking change we'll just bump the app version with a fix. For more context this method of having out of band updates adds a lot of maintenance work to the app and now puts pressure to keep old versions maintained as that out of band would be required to work across versions which means that out of band needs its own set of versions and the app would need to know which version(s) it's allowed to use. For example, say we make a change to some scripts out of band that are breaking in nature and only newer versions of the app supports that. Old versions now need to be aware of that and ensure that we keep a separate branch of versions for them.

**Comment by IndefiniteBen** [comment]
> > do you have a timeframe for it?
> 
> We're working on things we want to do before v2 fully releases and waiting on some upstream stuff then we'll probably ship it out. This app is a hobby project so we can't guarantee timeframes.

I think we all appreciate that, but can you give an estimated figure? How long do you expect those things and upstream stuff to take? Weeks? At least 2 months? 

Just trying to decide whether to try an RC or wait for the full release. Being very conservative, would you expect a full release this year? 

Thanks for all your work however long it takes.

**Comment by NovusTheory** [comment]
> Being very conservative, would you expect a full release this year?

Yes. The upstream is also estimated to do a new release within a week assuming that plan doesn't change.

**Comment by SirDufus** [comment]
I'm not sure if I need to post a bug as a new issue or in this thread, pardon me if I am reporting a shortcut bug here incorrectly.  Am running rc5 on Win11, all seems to work very well so far with exception of built in media keys on keyboard (Logi k400+) for play/pause, previous and next.

In previous versions, the built in media keys would always control the music player no matter what window was in focus. It no longer behaves this way and instead when using the media keys now, whatever current window is in focus is what the keys control. To be sure, I can reproduce it on Youtube.com, Netflix and Amazon prime. Once the player loses focus, I can't control the music player anymore until I make the window active again.

As this has never been the behavior on past versions, I am hoping it is a bug that can be easily fixed. Happy to do further testing / provide more feedback if it helps. Anyone else experiencing this?

**Comment by martinsolberg** [comment]
deb-package of rc5 works out of the box without any issues on Debian 12.5, even running in Wayland mode when the appropriate flags are passed to Electron.

**Comment by realSnosh** [comment]
how is it going?

**Comment by NovusTheory** [comment]
> how is it going?

There was a bit of a delay where v2 was expected to fully come out before 2024. Should be smooth sailing now.

**Comment by SirEndii** [comment]
Just want to yeet out some positive feedback

I used the old youtube music desktop v1 app one or two years ago, but I had several issues, so I removed it.
I found the v2 version today and I don't see any obvious errors or problems. The features I use work flawlessly 👍🏻

**Comment by EsekyIL** [comment]
A very nice project, I hope there will be a translation.
And you are very well done!

**Comment by sH1222J** [comment]
It's really unfortunate that there is no mini player function.

And I wish there were convenience functions related to housework.

**Comment by MattyGroch** [comment]
Hi friends!

I just found v2, and I noticed that there was a long-standing issue with the Last.fm integration in v1 that was resolved in v2. In v1, the album title was not included in the scrobble data, which had an impact when third-party apps would call Last.fm's API to pull recent track data. Thanks for getting this sorted out! I'm pumped for the full 2.0 launch.

**Comment by MattyGroch** [comment]
Oh no! Today I'm realizing there's no way to specify which sound device YTM Desktop uses for output. This is a huge bummer for me. Any chance that feature will make a comeback?

**Comment by Alipoodle** [comment]
> Oh no! Today I'm realizing there's no way to specify which sound device YTM Desktop uses for output. This is a huge bummer for me. Any chance that feature will make a comeback?

Yes. But the previous method was a huge "hack". You can do it natively in windows to select which audio source for a program to play through which will be more consistent then what we have as a temporary solution.
Not sure about other platforms as to if this is possible.

**Comment by gotgenes** [comment]
>Not sure about macOS, though.

I use and recommend [Rogue Amoeba's SoundSource](https://rogueamoeba.com/soundsource/) on macOS. It is not free (currently $39 USD) but it works well.

**Comment by gabuch2** [comment]
> Its trivial to do this on most Linux distros via Pulse Audio Volume Control, so there's no issue there either. Not sure about macOS, though.

Disagree, as any change of output device applied to the new ytmdesktop will be reflected on other Electron applications, as the system is unable to differentiate different Chromium processes. 

The current stable release, even with the hacky solution the devs mentioned was still consistent with the selected option, plus it didn't affect other Chromium applications like Discord, even though it was still identified as a Chromium instance in pavucontrol.

**Comment by Caboose12000** [comment]
(For context I'm on Nobara Linux using youtube-music-desktop-app-2.0.0.rc.5-1.x86_64.rpm) 
This may be a problem on my end since rc5 was working for me a week or two ago, but now it seems when I launch i get a black screen with a seek bar.  I can click the black space where the play/pause, next, back, buttons are and they seem to function.  It's playing songs that I had listened to last time I opened the app, but the main body of the page seems to be uninteractable.[
![ytm_blackscreen](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/99a83253-c4b6-48bc-b318-a6059aceee49)
](url)
has anyone else run into an issue like this?  I'm not sure what might've changed on my system to cause this new behavior.

**Comment by Caboose12000** [comment]
> > Its trivial to do this on most Linux distros via Pulse Audio Volume Control, so there's no issue there either. Not sure about macOS, though.
> 
> Disagree, as any change of output device applied to the new ytmdesktop will be reflected on other Electron applications, as the system is unable to differentiate different Chromium processes.

I'm curious what your setup is like that pulse audio volume control isn't working for you, for me it gives ytmd and discord the same label but can absolutely differentiate between them and I am able to set them to different output sources without issue.  
![pulse-audio-volume-control-works](https://github.com/ytmdesktop/ytmdesktop/assets/64852933/78a3facb-aa90-4a1b-a8ff-f9d019096593)
This may be a little off topic for this thread, so feel free to add me on discord if you wanna talk about it more @gabuch2 
my discord is caboose12000

**Comment by rafaelgomesxyz** [comment]
Thanks for the great work!

Is Last.fm not working right now? It asks me to allow the app, but then nothing happens and "User is Authenticated" shows "No".

**Comment by lordgabor** [comment]
> Thanks for the great work!
> 
> Is Last.fm not working right now? It asks me to allow the app, but then nothing happens and "User is Authenticated" shows "No".

For me it is working after I restarted the software (I am on Pop!_OS)

**Comment by IndefiniteBen** [comment]
> Is Last.fm not working right now? It asks me to allow the app, but then nothing happens and "User is Authenticated" shows "No".

I had the same. I think I just closed the settings window and when I opened it again it said "Yes". 
I may have restarted the app between authenticating, if refreshing the settings window doesn't fix it. 

Maybe a "refresh state" button could be next to the "User is Authenticated" text, to make it clearer to users? 

But this is getting fairly off-topic...

**Comment by Alipoodle** [comment]
Last.FM currently doesn't re-check that you've authenticated until a Restart has occurred OR you play a song as we don't run make another request to L.fm to check that the Key we just got from them is actually valid.

I might make add a fresh icon next to it to trigger the Integration to authenticate and update the value 👍

**Comment by FASouzaIT** [comment]
Any chance to get YTMDesktop Remote app for Android to work with v2 or get a new remote app for Android that works with companion server?

**Comment by umitseyhan75** [comment]
Is it possible to make Discord PRC look like "listening to yt music" instead of "playing yt music game"?

**Comment by NovusTheory** [comment]
> Is it possible to make Discord PRC look like "listening to yt music" instead of "playing yt music game"?
We already tell Discord to use a listening status. Discord rewrites it to playing.

**Comment by michalkobelka** [comment]
So, I've installed 2.0.1 version and I'm getting strange error and the app won't start:

```
YouTube Music Desktop App Crashed

Environment Details:
    YouTube Music Desktop App 2.0.1
    Electron 28.2.3
    Windows NT 10.0.22631

Name: TypeError
Message: Cannot read properties of undefined (reading 'videoId')
Cause: Unknown

TypeError: Cannot read properties of undefined (reading 'videoId')
    at IpcMainImpl.<anonymous> (C:\Users\uzivatel\AppData\Local\youtube_music_desktop_app\app-2.0.1\resources\app.asar\.webpack\main\index.js:8:415754)
    at IpcMainImpl.emit (node:events:517:28)
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:77582)
    at WebContents.emit (node:events:517:28)
```

I'm running Windows 11 on my PC. Any ideas how ti fix it? I've tried uninstalling Youtube Desktop Music, restarting PC, removing any appData\local\youutbe** data and no luck so far.

![ydm](https://github.com/ytmdesktop/ytmdesktop/assets/4164831/7209024a-9c04-451d-8d92-37540b9700ca)

**Comment by chriskol** [comment]
I've also started getting the above error on update.  I'm using the latest homebrew install on the latest osx if that's any help.

**Comment by NovusTheory** [comment]
Those of you experiencing this crash with videoId can you please provide more info in #1223 as that is the issue for tracking to help keep it centralized

**Comment by travisserio** [comment]
is there an active branch for the mini player? I sure miss that thing dearly..

**Comment by Alipoodle** [comment]
Closing this issue as v2 has been out for a while for now.
Most things have been implimented now into v2 version, but some which haven't been implimented are on our Radar still, but may not be a priority to get done over some other things we have in store. 🥰

### Some examples include
[Feature/YTMD-Remote](https://github.com/ytmdesktop/ytmdesktop/tree/v2-feature/ytmd-remote) - Is prep for the new version of remote which runs on the Companion server directly and not as an app.
[Feature/preferred-content-mode](https://github.com/ytmdesktop/ytmdesktop/tree/v2-feature/preferred-content-mode) (Req Premium) - Fixes the issue where YTM will swap from Video/Audio mode when the other unavalaible. This will help keep on the prefered method.
[Feature/Xosms](https://github.com/ytmdesktop/ytmdesktop/tree/v2-feature/xosms) - A Custom Media integrator for integration with Windows Media Service, mpris and MacOS* [MacOS support is in the works]
